packages/puppeteer-core/src/common/Puppeteer.ts

```diff
@@ -83,55 +133,4 @@ export class Puppeteer {
   connect(options: ConnectOptions): Promise<Browser> {
     return _connectToCDPBrowser(options);
   }
-
-  /**
-   * @deprecated Import directly puppeteer.
-   * @example
-   *
-   * ```ts
-   * import {registerCustomQueryHandler} from 'puppeteer';
-   * ```
-   */
-  registerCustomQueryHandler(
```

Comment:
Would not it be better if we keep this API and make custom query handlers to be instance specific?

Reply:
I wanted to do this, but wouldn't make sense with the current way puppeteer is used. Instance-specific query handles would imply you can have more than one instance of puppeteer, but this is not the case; you can't even create them. Even if we did have instance-specific APIs, I assume someone will comment asking for a global API and instance-specific ones like `Puppeteer.registerGlobalCustomQueryHandler` and `puppeteer.registerCustomQueryHandler`. In the end, there is a lot of feature-creep here. I prefer we just make it a global registry.

Reply:
Makes sense.